ibis/__init__.py

```diff
@@ -51,15 +53,29 @@
 __version__ = get_versions()['version']
 del get_versions
 
-for entry_point in pkg_resources.iter_entry_points(
-    group='ibis.backends', name=None
-):
-    try:
-        backend_module = entry_point.resolve()
-    except ImportError:
-        pass
-    else:
-        backend = backend_module.Backend()
-        setattr(ibis, entry_point.name, backend)
-        with ibis.config.config_prefix(entry_point.name):
-            backend.register_options()
+
+def __getattr__(name: str):
+    entry_points = list(
+        pkg_resources.iter_entry_points(group='ibis.backends', name=name)
+    )
+    if len(entry_points) == 0:
+        raise AttributeError(
+            f"module 'ibis' has no attribute '{name}'. "
+            f"If you are trying to access the '{name}' backend, "
+            f"try installing it first with `pip install ibis-{name}`"
+        )
+    elif len(entry_points) > 1:
+        warnings.warn(
+            f"More than one entrypoint found for backend '{name}', "
```

Comment:
Sorry not super familiar with this. What exactly mean to have more than one entrypoint for backend?

Reply:
(And why do we want to warn on this?)

Reply:
Imagine that two different developers create packages for bigquery. And both of them add an entrypoint with the name `bigquery`. When a user with both packages installed call `ibis.bigquery`, both will be found, but the first to be found is used, and the second will be ignored. @jreback suggests to raise an exception. I thought it'd make sense to move forward, knowing the user (with this warning) which of the packages will be used.

I'm ok raising an exception. And for your feedback, I think it makes sense to rephrase the message, to make it less technical, and more user friendly. Thanks for it.

Reply:
let's raise an exception. this is not tested at all in any event.

Reply:
I changed it, now it's raising an exception, and there is a test for it.